### PR TITLE
Index update to ES

### PR DESCRIPTION
**Pull request by dadepo**

### CODE REVIEW

**Comment by dadepo** [comment]
PS: Still looking at the failing tests. With the help of Timour, we found that it seems testcontainers do not work out of the box with Gitlab and we need to tweak the CI configuration. See [here](https://www.testcontainers.org/supported_docker_environment/continuous_integration/gitlab_ci/). Opening up this PR for comments while I look into doing just that.

File: whois-commons/src/main/java/net/ripe/db/whois/common/elasticsearch/IndexService.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Component
+public class IndexService {
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/elasticsearch/IndexService.java:34]: IndexService is really generic, can you rename to ```ElasticIndexService```

File: whois-commons/src/main/java/net/ripe/db/whois/common/elasticsearch/IndexMetadata.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package net.ripe.db.whois.common.elasticsearch;
+
+public class IndexMetadata {
```
Comment (eshryane) [inline whois-commons/src/main/java/net/ripe/db/whois/common/elasticsearch/IndexMetadata.java:3]: Rename to ElasticIndexMetadata

File: whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/IndexServiceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@org.junit.jupiter.api.Tag("IntegrationTest")
+public class IndexServiceIntegrationTest {
```
Comment (eshryane) [inline whois-api/src/test/java/net/ripe/db/whois/api/elasticsearch/IndexServiceIntegrationTest.java:31]: Rename to ElasticIndexServiceIntegrationTest

File: whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ESFullTextIndex.java [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+// TODO [DA] Lucene implementation has some mechanism around thread safety. check if that is also necessary
+@Component
+public class ESFullTextIndex {
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ESFullTextIndex.java:33]: Rename to ElasticFullTextIndex (Elastic is more readable to me than ES)

File: whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticFullTextIndex.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+// TODO [DA] Lucene implementation has some mechanism around thread safety. check if that is also necessary
+@Component
+public class ElasticFullTextIndex {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ElasticFullTextIndex.class);
+    private ElasticIndexService elasticIndexService;
+    private final JdbcTemplate jdbcTemplate;
+    private final String source;
+    private final String TASK_NAME = "fulltextIndexUpdate";
+
+    public ElasticFullTextIndex(ElasticIndexService elasticIndexService,
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticFullTextIndex.java:40]: Missing Autowired annotation on constructor?

**Review by eshryane — APPROVED** [review]

**Pull request merged**